libs/community/langchain_community/utilities/jira.py

```diff
@@ -14,6 +14,7 @@ class JiraAPIWrapper(BaseModel):
     confluence: Any = None
     jira_username: Optional[str] = None
     jira_api_token: Optional[str] = None
+    jira_oauth_dict: Optional[Union[Dict[str, Union[str, Dict[str, str]]], str]] = None
```

Comment:
1. `dict` isn't a good thing to put in this variable name. Perhaps `jira_oauth2` to indicate it's just proxying the `oauth2` variable?
2. Could you document inline what values are accepted? And explicit type annotation for the dict using a TypedDict (`typing_extensions import TypedDict`) would help!
3. Now that there's more than one option to authenticate, it would be good to document this as part of the wrapper -- you could add an authentication example in the JiraAPIWrapper and inline a doc-string for both variables

Reply:
Is it what you had in mind?